### DISCUSSION TITLE
How to find the installed Dex version ?

### DISCUSSION THREAD

**Discussion by iam-veeramalla** [dexidp/dex#2334] [Q&A] [answered]
Hi , We run a productized Image of Dex. What is the easiest way to identify the installed version of Dex ? I expect Dex to print the version in the first few lines of logs as many other applications does. 

Is it already available in any files or somewhere ?

**Comment by nabokihms** [comment]
Currently, there are two ways to determine the Dex version.
1. Execute `dex version` inside the Pod.
2. Request version via GRPC API.

I see no obstacles printing the version of Dex first line in the log.

**Reply by iam-veeramalla** [reply]
Hi @nabokihms , 
 This is what I receive when I run `dex version`
 ```
 sh-4.4$ dex version
Dex Version: DEV
Go Version: go1.16.6
Go OS/ARCH: linux amd64
 ```

**Reply by nabokihms** [reply]
Is this an official Dex image?

**Reply by iam-veeramalla** [reply]
No. We at Red Hat only use productized Images. we make some modifications to the actual Dockerfile.

**Reply by iam-veeramalla** [reply]
Can you tell me, who to achieve this ?
`Request version via GRPC API.`

**Reply by sagikazarmark** [reply]
When you build the Dex binary, you can pass the following to the Go build command to add your own version: `-ldflags "-X main.version=YOUR_VERSION". Then `dex version` should print the correct version.

The gRPC API is described here: https://github.com/dexidp/dex/blob/master/api/v2/api.proto#L180
It can be enabled in the configuration, but it uses the same variable for determining the version, so unless you inject it yourself, you won't see it.

**Reply by iam-veeramalla** [reply]
@sagikazarmark Thanks a-lot for the explanation. Do you also think it is useful to print the version of Dex in the logs ? If yes, probably I can raise a PR.

**Reply by sagikazarmark** [reply]
It would probably make sense to print the version in the log when Dex starts, yes.

**Comment by iam-veeramalla** [comment, marked as answer]
Closing the discussion. Thanks @sagikazarmark  @nabokihms for helping me out. 

1. I will add the build flag to the Dockerfile as suggested. 
2. I will create an issue to print the version of Dex in the logs.
